test/Libraries/AnalysisTests/LabelTests.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using Analysis;
+using Autodesk.DesignScript.Geometry;
+using Autodesk.DesignScript.Interfaces;
+using Dynamo.Visualization;
+using NUnit.Framework;
+
+namespace AnalysisTests
+{
+    [TestFixture]
+    class LabelTests
+    {
+        /// <summary>
+        /// This test method will execute the next methods from the Label class:
+        /// private Label(Point point, string label)
+        /// public static Label ByPointAndString(Point point, string label)
+        /// public void Tessellate(IRenderPackage package, TessellationParameters parameters)
+        /// </summary>
+        [Test, Category("UnitTests")]
+        public void LabelByPointAndStringTest()
+        {
+            //Arrange
+            var labelTest = Label.ByPointAndString(Point.ByCoordinates(100,100), "TestingLabel");
+            var factory = new DefaultRenderPackageFactory();
+            var package = factory.CreateRenderPackage();
+
+            //Act
+            labelTest.Tessellate(package, new TessellationParameters());
```

Comment:
What are you testing from the `Tessellate` call? I think the assertions below will pass even if you don't call `Tessellate`, wouldn't they?

Reply:
@aparajit-pratap 
The Tessellate method is a [IsVisibleInDynamoLibrary(false)], has no code coverage and is not used (inside the Label class o in other classes, event the whole Label class is not used), I was proposing to delete the whole Label.cs file but Michael said the next "if it's public, we cannot get rid of it as it will break semantic versioning."
You are correct, the assertions are checking the results of Label.ByPointAndString method then they don't depend of calling the  Tessellate method, then is it ok if I just delete the call to the Tessellate method (of course if won't have coverage)?

Reply:
It looks like the package properties are modified in Tessellate. You can check for those properties after the call to Tessellate.

Reply:
@aparajit-pratap this test method was re-written due to some comments that Michael did, please review again.